ichnaea/api/locate/tests/test_query.py

```diff
@@ -352,63 +343,58 @@ def _make_query(
             cell=self.cell_model_query(cell),
             wifi=self.wifi_model_query(wifi),
             geoip_db=geoip_db,
-            stats_client=stats,
             **kw,
         )
         query.emit_query_stats()
         return query
 
-    def test_no_log(self, geoip_db, stats):
+    def test_no_log(self, geoip_db, metricsmock):
         api_key = KeyFactory(valid_key=None)
-        self._make_query(geoip_db, stats, api_key=api_key, api_type="locate")
-        stats.check(total=0)
-
-    def test_empty(self, geoip_db, stats):
-        self._make_query(geoip_db, stats, ip=self.london_ip)
-        stats.check(
-            counter=[
-                (
-                    "locate.query",
-                    ["key:test", "region:GB", "blue:none", "cell:none", "wifi:none"],
-                )
-            ]
+        self._make_query(geoip_db, api_key=api_key, api_type="locate")
+        assert len(metricsmock.get_records()) == 0
+
+    def test_empty(self, geoip_db, metricsmock):
+        self._make_query(geoip_db, ip=self.london_ip)
+        assert metricsmock.has_record(
+            "incr",
+            "locate.query",
+            value=1,
+            tags=["key:test", "region:GB", "blue:none", "cell:none", "wifi:none"],
         )
 
-    def test_one(self, geoip_db, stats):
+    def test_one(self, geoip_db, metricsmock):
         blues = BlueShardFactory.build_batch(1)
         cells = CellShardFactory.build_batch(1)
         wifis = WifiShardFactory.build_batch(1)
 
         self._make_query(
-            geoip_db, stats, blue=blues, cell=cells, wifi=wifis, ip=self.london_ip
-        )
-        stats.check(
-            total=1,
-            counter=[
-                (
-                    "locate.query",
-                    ["key:test", "region:GB", "blue:one", "cell:one", "wifi:one"],
-                )
-            ],
-        )
+            geoip_db, blue=blues, cell=cells, wifi=wifis, ip=self.london_ip
+        )
+        assert metricsmock.get_records() == [
```

Comment:
Since the previous check makes sure there's only one of these `incr` records, I converted it to this `get_records()` assertion which both asserts that there's only one record and it looks like this.